process-migration-service/src/main/java/org/kie/processmigration/service/impl/KieServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.kie.processmigration.model.KieServerConfig;
+import org.kie.processmigration.model.exceptions.InvalidKieServerException;
+import org.kie.processmigration.service.KieService;
+import org.kie.server.api.marshalling.MarshallingFormat;
+import org.kie.server.client.CredentialsProvider;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.KieServicesFactory;
+import org.kie.server.client.QueryServicesClient;
+import org.kie.server.client.admin.ProcessAdminServicesClient;
+import org.kie.server.client.credentials.EnteredCredentialsProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wildfly.swarm.spi.api.config.ConfigView;
+
+@ApplicationScoped
+public class KieServiceImpl implements KieService {
+
+    private static final String HOST = "host";
+    private static final String PORT = "port";
+    private static final String PROTOCOL = "protocol";
+    private static final String CONTEXT_ROOT = "contextRoot";
+    private static final String PATH = "path";
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+
+    private static final Logger logger = LoggerFactory.getLogger(KieServiceImpl.class);
+
+    private Map<String, KieServerConfig> configs = new HashMap<>();
+    private Map<String, KieServicesClient> kieServices = new HashMap<>();
+
+    @Inject
+    private ConfigView configView;
+
+    @SuppressWarnings("unchecked")
+    public void loadConfigs(@Observes @Initialized(ApplicationScoped.class) Object event) {
+        List<Map<String, String>> value = configView.resolve("kieservers").as(List.class).getValue();
+        value.stream().forEach(this::loadConfig);
+    }
+
+    public Map<String, KieServerConfig> getConfigs() {
+        return configs;
+    }
+
+    public ProcessAdminServicesClient getProcessAdminServicesClient(String kieServerId) throws InvalidKieServerException {
+        if (!kieServices.containsKey(kieServerId)) {
+            throw new InvalidKieServerException(kieServerId);
+        }
+        return (ProcessAdminServicesClient) kieServices.get(kieServerId).getServicesClient(ProcessAdminServicesClient.class);
+    }
+
+    public QueryServicesClient getQueryServicesClient(String kieServerId) throws InvalidKieServerException {
+        if (!kieServices.containsKey(kieServerId)) {
+            throw new InvalidKieServerException(kieServerId);
+        }
+        return (QueryServicesClient) kieServices.get(kieServerId).getServicesClient(QueryServicesClient.class);
+    }
+
+    private void loadConfig(Map<String, String> config) {
+        KieServerConfig kieConfig = new KieServerConfig();
```

Comment:
Yes, in the configuration file

Reply:
so that means to add another kie server we need to bounce the application so the config is reloaded, right? Just wondering how we can prevent users from defining migration plans for particular server that might not exist...

Reply:
Yes, a configuration change would require the application to restart. We can work on some kind of discovery service using K8S API but this can be added later on.
In the UI it "should" be easy as the user will only be able to select from the existing instances.
Additionally maybe a "validate" or "dry-run" operations can help with that. WDYT?

Reply:
I'd say the service should reject migration plans (or put them in some sort of locked state) for kie servers it does not manage

Reply:
We can reject migration plans at creation/update and reject migrations when they're submitted in case the process existed when the plan was created but it was removed afterwards.

Reply:
I'd say we need to make this at both places - first at creation/update time so that essentially will discard particular migration plan and on execution time it should fail the migration with proper message.

Reply:
I have done the validation at the time of creating an async migration and when starting the migration. 
As plans are not related to any kieserver, the validation cannot be done.